syntax = "proto3";

import 'files/common/common.proto';

message RecurringRewardsLists {
  repeated RecurringRewardsList recurring_rewards_lists = 1;

  message RecurringRewardsList {
    ObjectIdentity identity = 1;
    RewardsInfo info = 2;
  }

  message RewardsInfo {
    string name_placeholder = 1;
    int32 tier = 4;
    int32 limit = 5;
    int32 collection_interval_seconds = 6;
    int32 num_collections = 7;
    repeated RewardItem items = 9;
    repeated RewardItemAlt items_alt = 13;
  }

  message RewardItem {
    int64 id = 3;
    int32 quality = 5;
    int32 quantity = 6;
  }

  message RewardItemAlt {
    ObjectIdentity identity = 1;
    int32 quantity = 2;
  }
}